editoast/src/modelsv2/connection_pool/db_connection_pool.rs

```diff
@@ -0,0 +1,92 @@
+use diesel_async::pooled_connection::deadpool::Object;
+use diesel_async::pooled_connection::deadpool::Pool;
+
+use diesel_async::AsyncPgConnection;
+use std::sync::Arc;
+#[cfg(test)]
+use tokio::sync::OwnedRwLockWriteGuard;
+#[cfg(test)]
+use tokio::sync::RwLock;
+#[cfg(test)]
+use url::Url;
+
+use super::DbConnectionError;
+
+/// Wrapper for connection pooling with support for test connections.
+/// In test mode, the connection pool will always provide the same test connection for testing purposes.
+/// This connection will not commit any changes to the database, ensuring the isolation of each test.
+///
+/// # Test creation steps exemple
+/// ## Create the connection pool in test mode:
+///     let pool = create_test_connection_pool().await;
+/// ## Create the service with the connection pool:
+///     let service = create_test_service_with_connection_pool(pool.clone()).await;
+/// ## Insert the required data before testing the API service:
+///     let document = create_document(&pool).await;
+/// ## Call the API service
+/// ## Test the API service response
+///
+/// The connection pool in test mode will not commit any changes to the database,
+/// all data will be deleted after the test
+#[derive(Clone)]
+pub struct DbConnectionPoolV2 {
+    pub pool: Arc<Pool<AsyncPgConnection>>,
+    #[cfg(test)]
+    test_connection: Option<Arc<RwLock<Object<AsyncPgConnection>>>>,
+}
+
+#[cfg(test)]
+impl DbConnectionPoolV2 {
+    pub async fn create(pool: Pool<AsyncPgConnection>) -> Result<Self, DbConnectionError> {
+        use diesel_async::AsyncConnection;
+        let mut conn = pool.get().await.map_err(DbConnectionError::DeadpoolPool)?;
```

Comment:
```suggestion
        let mut conn = pool.get().await?;
```

With `#[from]` annotations in the error enum this may be done implicitly.